src/network/index.js

```diff
@@ -1,68 +1,82 @@
 'use strict'
 
 const bl = require('bl')
-const fs = require('fs')
-const path = require('path')
-const protobuf = require('protocol-buffers')
-const pbm = protobuf(fs.readFileSync(path.join(__dirname, '../message/message.proto')))
+const async = require('async')
+const Message = require('../message')
 
 module.exports = class Network {
   constructor (libp2p, peerBook, bitswap) {
     this.libp2p = libp2p
     this.peerBook = peerBook
+    this.bitswap = bitswap
 
-    this.libp2p.swarm.handle('/ipfs/bitswap/1.0.0', (conn) => {
-      conn.pipe(bl((err, data) => {
-        conn.end()
-        if (err) {
-          return bitswap._receiveError(err)
-        }
-        let msg
-        try {
-          msg = pbm.Message.decode(data)
-        } catch (err) {
-          return bitswap._receiveError(err)
-        }
-        bitswap._receiveMessage(conn.peerId, msg)
-      }))
-    })
+    this._attachSwarmListeners()
+  }
 
-    this.libp2p.swarm.on('peer-mux-established', (peerInfo) => {
-      bitswap._onPeerConnected(peerInfo.id)
-    })
+  _attachSwarmListeners () {
+    this.libp2p.swarm.handle('/ipfs/bitswap/1.0.0', this._onConnection.bind(this))
 
-    this.libp2p.swarm.on('peer-mux-closed', (peerInfo) => {
-      bitswap._onPeerDisconnected(peerInfo.id)
-    })
+    this.libp2p.swarm.on('peer-mux-established', this._onPeerMux.bind(this))
```

Comment:
This changes what gets passed to the function to become a peerInfo instead of a peerId (as you had specified on the interface/stub)